### ISSUE TITLE
Make linter to prevent usage of native operators for sdk values

### ISSUE THREAD

**Issue by czarcas7ic** [osmosis-labs/osmosis#5963]
## Background

When utilizing sdk values (sdk.Int, sdk.Dec, etc), you need to use sdk operators (.Equal(), .GT(), etc) instead of native operators (==, >, etc). Its very easy to overlook this and waste a lot of time debugging when in reality you were using the wrong operator type. We should make a linter for this.

## Suggested Design

Dev made a suggestion of adding it into codeQL.

## Acceptance Criteria

When using native operators, they either get flagged by the linter, or the linter automatically replaces it with the respective sdk operator.

**Comment by AlpinYukseloglu** [comment]
If we're only going to be setting up code analysis for this check, I wonder if it would be simpler to just have a go AST based script that runs in CI for this?

We'd lose some of the extensibility of codeQL but imo would be simpler to add especially since it seems we only intend to use this in CI and not as part of github's security interface

**Comment by ValarDragon** [comment]
yeah, and theres a number of nice add-ons to CodeQL in the SDK that I feel like we should be adding
